src/PageContent/RequestPickup/RecurringPickupRequest.js

```diff
@@ -0,0 +1,284 @@
+        });
+    }
+
+    // write to firebase
+    // var newRequest = firebase.database().ref().child("delivery_requests").push();
+    // newRequest.set(deliveryRequest);
+    
+    submitRequest(){
+        
+    }
+
+    createRequest(event) {
+        
+        if(!this.handleValidation()){ 
+            event.preventDefault();
+            alert('Form has errors');
+        }else{
+            event.preventDefault();
+            var raUid = event.target.receiveingAgency.value;
+            if (!raUid) raUid = null;
+            
+            var dgUid = event.target.delivererGroup.value;
+            if (!dgUid) dgUid = null;
+
+            var durationValue = null;
+            if (event.target.durationType.value === RequestDurationType.DATE) {
+                durationValue = event.target.endDate.value;
+            } else {
+                durationValue = event.target.numRecurrences.value;
+            }
+
+            // create DeliveryRequest object
+            var deliveryRequest = {
+                startDate: event.target.startDate.value,
+                duration:{
+                    type: event.target.durationType.value,
+                    value: durationValue
+                },
+                repeats: event.target.repeats.value,
+                startTime: event.target.startTime.value,
+                endTime: event.target.endTime.value,
+                primaryContact: event.target.primaryContact.value,
+                notes: event.target.notes.value,
+                receiveingAgency: {
+                    uid: raUid,
+                    confirmed: false
+                },
+                delivererGroup: {
+                    uid: dgUid,
+                    confirmed: false
+                },
+                requestTimestamp: Date.now()
+            };
+            let date = new Date(deliveryRequest.startDate);
+            let day = date.getDay();
+            let days = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'];
+
+            this.setState({
+                formInfo: deliveryRequest,
+                dayOfWeek: days[day]
+            });
+            
+            this.toggleModal();
+        }
+    }
+
+    render() {
+        return (
+            <div className="form" id="recurringForm">
+                <form onSubmit={this.createRequest}>
+                    <div className="info">
+                        <p id="form-heading">Schedule Recurring Pickup</p>
+                        <p className="error">{this.state.errors['endBeforeStart']}</p>
+                        <p className="error">{this.state.errors['startTime']}</p>
+                        <p className="error">{this.state.errors['endTime']}</p>
+                        <p className="error">{this.state.errors['recurTimes']}</p>
+                        <p className="error">{this.state.errors['startDate']}</p>
+                        <p className="error">{this.state.errors['time']}</p>
```

Comment:
add to TODO: loop through everything in `this.state.errors` rather than hardcoding - could be error-prone, easily miss one when new checks are added